### PR TITLE
feat: implement checkpoint persistence system for LangGraph workflows

### PR DESCRIPTION
**Pull request by MH4GF**
## Issue

- https://github.com/liam-hq/liam/pull/2926
    - Check here to see the overall design.
- Implement Steps
    - [ ] Implementing SupabaseSaver ← now
    - [ ] Use SupabaseSaver as a checkpoint in graph
    - [ ] Implement session resume in executeDesignProcess as a result of checkpoint
    - [ ] Delete history from workflow state as a result of checkpoint
    - [ ] Switch workflow execution from trigger.dev back to Vercel

## Why is this change needed?

This PR implements a complete checkpoint persistence system for LangGraph workflows using Supabase as the storage backend. The implementation provides persistent state management with organization-level data isolation for multi-tenancy support.

### Key features
- Full implementation of SupabaseCheckpointSaver extending BaseCheckpointSaver
- Organization-level data isolation for multi-tenancy  
- Support for checkpoint hierarchies with parent-child relationships
- Channel value persistence with efficient blob storage
- Pending write handling for workflow state updates
- Comprehensive test coverage with MSW for API mocking

### Database schema
Added three new tables to support checkpoint persistence:
- `checkpoints` - Core checkpoint metadata and relationships
- `checkpoint_blobs` - Channel values stored as binary data  
- `checkpoint_writes` - Pending write operations

### Technical implementation
- BYTEA column handling with Base64 encoding for Supabase REST API
- Efficient serialization/deserialization using LangGraph's built-in serializer
- Proper error handling for missing configurations
- Type-safe interfaces with runtime validation using Valibot

## Test plan
- [x] Comprehensive unit tests covering all checkpoint operations
- [x] Mock server setup using MSW for isolated testing
- [x] Test coverage for edge cases and error scenarios
- [x] Organization data isolation verification
- [x] All tests passing with `pnpm test`
- [x] Linting and type checking pass

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* **New Features**
  * Introduced robust checkpoint persistence for workflows, enabling session resumption and state management.
  * Added support for storing, retrieving, listing, and versioning checkpoints with organization-level data isolation.
  * Implemented validation schemas for checkpoint data and metadata to ensure data integrity.
  * Comprehensive test suite added to verify checkpoint operations and error handling.

* **Database**
  * Added four new tables for checkpoint storage, blobs, writes, and migrations, with multi-tenant security and indexing.
  * Enabled row-level security policies and triggers for data integrity and organizational isolation.

* **Chores**
  * Updated development dependencies to include the `msw` library for improved testing capabilities.

### CODE REVIEW

File: frontend/internal-packages/agent/src/checkpoint/SupabaseCheckpointSaver.ts
Comment (MH4GF) [inline frontend/internal-packages/agent/src/checkpoint/SupabaseCheckpointSaver.ts:1]: The following PostgresSaver was used as a reference and adjusted to be handled by supabase-js.

https://github.com/langchain-ai/langgraphjs/blob/c4ffea4371eab1ad9daae970a5ede6b0a676e6c4/libs/checkpoint-postgres/src/index.ts

File: frontend/internal-packages/agent/src/checkpoint/SupabaseCheckpointSaver.test.ts
Comment (MH4GF) [inline frontend/internal-packages/agent/src/checkpoint/SupabaseCheckpointSaver.test.ts:1]: Since I did not want to mock with `v.fn()`, we used msw to achieve mocking at the HTTP request level. This greatly increased the scope of testing.

File: frontend/internal-packages/db/supabase/migrations/20250806082517_add_checkpoint_tables.sql [hunk 1]
```diff
@@ -0,0 +1,511 @@
...
+GRANT ALL ON TABLE "public"."checkpoint_writes" TO "authenticated";
+GRANT ALL ON TABLE "public"."checkpoint_writes" TO "service_role";
+
+GRANT ALL ON TABLE "public"."checkpoint_migrations" TO "anon";
+GRANT ALL ON TABLE "public"."checkpoint_migrations" TO "authenticated";
+GRANT ALL ON TABLE "public"."checkpoint_migrations" TO "service_role";
+
+GRANT ALL ON FUNCTION "public"."update_checkpoints_updated_at"() TO "anon";
+GRANT ALL ON FUNCTION "public"."update_checkpoints_updated_at"() TO "authenticated";
+GRANT ALL ON FUNCTION "public"."update_checkpoints_updated_at"() TO "service_role";
```
[the remark this answers is not in the document]
Comment (MH4GF) [reply frontend/internal-packages/db/supabase/migrations/20250806082517_add_checkpoint_tables.sql:509]: That's right. All permissions of anon users will be revoked prior to release. Once this is done, we'll leave it as is.

File: frontend/internal-packages/db/schema/schema.sql [hunk 2]
```diff
@@ -1120,6 +1133,144 @@ CREATE TABLE IF NOT EXISTS "public"."building_schemas" (
...
+    "id" "uuid" DEFAULT "gen_random_uuid"() NOT NULL,
+    "thread_id" "text" NOT NULL,
+    "checkpoint_ns" "text" DEFAULT ''::"text" NOT NULL,
+    "checkpoint_id" "text" NOT NULL,
+    "parent_checkpoint_id" "text",
+    "checkpoint" "jsonb" NOT NULL,
+    "metadata" "jsonb" DEFAULT '{}'::"jsonb" NOT NULL,
+    "organization_id" "uuid" NOT NULL,
+    "created_at" timestamp with time zone DEFAULT "now"(),
+    "updated_at" timestamp with time zone DEFAULT "now"()
```
Comment (FunamaYukina) [inline frontend/internal-packages/db/schema/schema.sql:1235]: [ask]
I think only the `checkpoints` table has `updated_at`, is there a case where the record is updated?👀

**Review by FunamaYukina — APPROVED** [review]
I think it looks good!
I asked one question.

File: frontend/internal-packages/db/schema/schema.sql (hunk 2 above)
Comment (MH4GF) [reply frontend/internal-packages/db/schema/schema.sql:1235]: Yes, SupabaseSaver is upsert here.

https://github.com/liam-hq/liam/pull/2935/files#diff-49827aeea0324bd74ff993b5862788ee14afd313155894efbd68889e77bb989eR336-R338
